docs/go/signals.md

```diff
@@ -72,6 +72,39 @@ if err != nil {
 
 You can find more example Signals usage in our [Go Samples](https://github.com/temporalio/samples-go).
 
+## Signalling with structs
+
+You can also send structs in signals, as long as the struct is [serializable](https://pkg.go.dev/go.temporal.io/sdk/converter#CompositeDataConverter.ToPayload).
+`Receive()` decodes data into a generic map, to decode the value into a struct you should use a library like [mapstructure](https://github.com/mitchellh/mapstructure).
+Note that Temporal only serializes public fields.
+
+```go
+// Make sure all fields you want to serialize are public. Temporal
+// serializes `Message`, not `message`.
```

Comment:
> Temporal serializes 'Message', not 'message'

Could this comment be generalized to mention how this applies for any struct field? (serialized "as-is"/"preserving the case"/....)